searches/linear_search.nim

```diff
@@ -0,0 +1,85 @@
+## Linear Search
+## =============
+## Linear search is the simplest but least efficient algorithm
+## to search for an element in a data set.
+## It examines each element until it finds a match,
+## starting at the beginning of the data set, until the end.
+## The search is finished and terminated once the target element is located.
+## https://www.simplilearn.com/tutorials/data-structure-tutorial/linear-search-algorithm
+##
+## Time Complexity: O(N)
+## Space Complexity in for-loop linear search: O(1)
+## Space Complexity in recursive linear search: O(n)
+## This is because of the call stack operation,
+## and there is no CPS automatic optimization in Nim
+## CPS: https://en.wikipedia.org/wiki/Continuation-passing_style
+
+runnableExamples:
+  var arr1 = [0, 3, 1, 4, 5, 6]
+  doAssert linearSearch(arr1, 5) == some(Natural(4))
+  doAssert recursiveLinearSearch(arr1, arr1.high, 5) == some(Natural(4))
+
+  var arr2 = ['0', 'c', 'a', 'u', '5', '7']
+  doAssert linearSearch(arr2, '5') == some(Natural(4))
+  doAssert recursiveLinearSearch(arr2, arr2.high, '5') == some(Natural(4))
+
+
+## importing options and system for type Option and Natural
```

Comment:
```suggestion
```
It's self-explaining. If you wish to leave it, make it a regular comment (with a single hash), otherwise it will be displayed as a free-standing sentence in the documentation.